### PR TITLE
issue(#2449) fixed 

### LINKED ISSUE #2449: Crashed when uploading an image to server which is in read-only mode
Whenever I am trying to upload an image from home page the app is getting crashed

### PR DESCRIPTION
**Pull request by aniketnath**
Fixed #2449 Crashed when uploading an image to the server which is in read-only mode

What changes did you make and why?

It was getting a null pointer exception

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/mwapi/UploadResult.java [hunk 1]
```diff
@@ -36,13 +36,17 @@
...
     @Override
     public String toString() {
-        return "UploadResult{" +
-                "errorCode='" + errorCode + '\'' +
-                ", resultStatus='" + resultStatus + '\'' +
-                ", dateUploaded='" + dateUploaded.toString() + '\'' +
-                ", imageUrl='" + imageUrl + '\'' +
-                ", canonicalFilename='" + canonicalFilename + '\'' +
-                '}';
+        if (dateUploaded != null) {
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/mwapi/UploadResult.java]: Do not add the check here. Add the check just for `dateuploaded`.

Comment (aniketnath) [reply app/src/main/java/fr/free/nrw/commons/mwapi/UploadResult.java]: ok

Comment (aniketnath) [reply app/src/main/java/fr/free/nrw/commons/mwapi/UploadResult.java]: I have added checking only for dateUpdate. check commit no 3

File: app/src/main/java/fr/free/nrw/commons/mwapi/UploadResult.java [hunk 2]
```diff
@@ -36,13 +36,16 @@
...
     @Override
     public String toString() {
-        return "UploadResult{" +
-                "errorCode='" + errorCode + '\'' +
-                ", resultStatus='" + resultStatus + '\'' +
-                ", dateUploaded='" + dateUploaded.toString() + '\'' +
-                ", imageUrl='" + imageUrl + '\'' +
-                ", canonicalFilename='" + canonicalFilename + '\'' +
-                '}';
+        if (dateUploaded == null) {
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/mwapi/UploadResult.java]: No not return early if date uploaded is null. 

add a check like: `dateUploaded==null? "":dateUploaded.toString()`

File: app/src/main/java/fr/free/nrw/commons/mwapi/UploadResult.java [hunk 3]
```diff
@@ -36,13 +36,16 @@
...
+            return "Server is in read only mode";
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/mwapi/UploadResult.java]: Do not return early.

File: app/src/main/res/drawable/ic_info_outline_black_24dp.xml [hunk 4]
```diff
@@ -4,6 +4,6 @@
         android:viewportWidth="24.0"
         android:viewportHeight="24.0">
     <path
-        android:fillColor="#FF000000"
+        android:fillColor="#FFFFFF"
```
Comment (maskaravivek) [inline app/src/main/res/drawable/ic_info_outline_black_24dp.xml]: Undo unrelated changes

File: app/src/main/java/fr/free/nrw/commons/mwapi/UploadResult.java (hunk 2 above)
Comment (aniketnath) [reply app/src/main/java/fr/free/nrw/commons/mwapi/UploadResult.java]: I have changed the condition checking into a single line. Check the last commit.

File: app/src/main/res/drawable/ic_info_outline_black_24dp.xml (hunk 4 above)
Comment (aniketnath) [reply app/src/main/res/drawable/ic_info_outline_black_24dp.xml]: I am new in open source contribution. How can I remove this commit?

Comment (maskaravivek) [reply app/src/main/res/drawable/ic_info_outline_black_24dp.xml]: You can revert a commit by using `git revert <commit sha>` or else you can manually undo the changes and commit it. :)

Comment (aniketnath) [reply app/src/main/res/drawable/ic_info_outline_black_24dp.xml]: I am creating a new pull request by doing fresh commit

Comment (maskaravivek) [reply app/src/main/res/drawable/ic_info_outline_black_24dp.xml]: Sure. Please close this pull request once your new PR is up. :)

Comment (aniketnath) [reply app/src/main/res/drawable/ic_info_outline_black_24dp.xml]: yea sure
